terraform-provider-constellation/internal/provider/image_data_source.go

```diff
@@ -128,11 +128,11 @@ func (d *ImageDataSource) ValidateConfig(ctx context.Context, req datasource.Val
 		)
 	}
 
-	// Marketplace image is only supported for Azure
-	if !data.CSP.Equal(types.StringValue("azure")) && !data.MarketplaceImage.IsNull() {
+	// Marketplace image is only supported for Azure and GCP
+	if data.CSP.Equal(types.StringValue("aws")) && !data.MarketplaceImage.IsNull() {
 		resp.Diagnostics.AddAttributeWarning(
 			path.Root("marketplace_image"),
-			"Marketplace images are currently only supported on Azure", "When another CSP than Azure is used, setting 'marketplace_image' has no effect.",
+			"Marketplace images are currently only supported on Azure and GCP", "When another CSP than Azure or GCP is used, setting 'marketplace_image' has no effect.",
```

Comment:
Is "has no effect" actually true? Reading the code, it seems that it will fail with an error instead. (Haven't tried.)

Reply:
You're correct. Since the MPI refactorings, this is not the case anymore.